### PR TITLE
Merge CreateTLDUidsIterator into CreateUidsIterator

### PR DESCRIPTION
**Pull request by apmoriarty**
Moved functionality from CreateTLDUidsIterator into CreateUidsIterator. Any query can now parse root TLD uids by enabling the parseRootUids option.

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/index/lookup/RangeStream.java [hunk 1]
```diff
@@ -457,12 +456,7 @@ public ScannerStream visit(ASTEQNode node, Object data) {
                 
                 if (setCondenseUids) {
```
Comment (jwomeara) [inline warehouse/query-core/src/main/java/datawave/query/index/lookup/RangeStream.java]: Is setCondenseUids synonymous with parseRootUids?

File: warehouse/query-core/src/main/java/datawave/query/index/lookup/CreateUidsIterator.java [hunk 2]
```diff
@@ -62,8 +70,10 @@
     private static final Logger log = Logger.getLogger(CreateUidsIterator.class);
     
     public static final String COLLAPSE_UIDS = "index.lookup.collapse";
+    public static final String PARSE_ROOT_UIDS = "index.lookup.parse.root.uids";
```
Comment (jwomeara) [inline warehouse/query-core/src/main/java/datawave/query/index/lookup/CreateUidsIterator.java]: This turns TLD parsing on or off, right?  So why not name it TLD... something?

Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/index/lookup/CreateUidsIterator.java]: I can change that to PARSE_TLD_UIDS

File: warehouse/query-core/src/main/java/datawave/query/index/lookup/RangeStream.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/index/lookup/RangeStream.java]: No, not exactly...

**Comment by apmoriarty** [comment]
> 
> 
> I would be in favor of having 'TLD' somewhere in the name the setting that turns this on/off. Thoughts?
> 
> Also, do we have any tests which verify that everything still works in the non-TLD case?

I can rename variables/methods from "parse root uids" to "parse tld uids". Queries run as non-tld by default, so yes we're covered in the non-tld case.

**Pull request merged**